generate/generate.go

```diff
@@ -59,44 +60,67 @@ func BuildStructs(schemaPath string, outputDir string) error {
 		name := strings.Split(filepath.Base(allOfPath), ".")[0]
 		embeds = append(embeds, exportedName(name))
 
-		if err := buildStructFile(schemaPath, allOfPath, name, packageName, nil, outputDir); err != nil {
-			return fmt.Errorf("failed to build struct file for %q: %v", name, err)
+		if err := buildFiles(schemaPath, allOfPath, name, packageName, nil, outputDir, useMessagePack); err != nil {
+			return err
 		}
 	}
 
 	for _, oneOfPath := range oneOfTypes {
 		name := strings.Split(filepath.Base(oneOfPath), ".")[0]
 
-		if err := buildStructFile(schemaPath, oneOfPath, name, packageName, embeds, outputDir); err != nil {
-			return fmt.Errorf("failed to build struct file for %q: %v", name, err)
+		if err := buildFiles(schemaPath, oneOfPath, name, packageName, embeds, outputDir, useMessagePack); err != nil {
+			return err
 		}
 	}
 
 	return nil
 }
 
-func buildStructFile(schemaPath, childPath, name, packageName string, embeds []string, outputDir string) error {
+// Build specified struct file and (if enabled) MessagePack serialization methods.
+func buildFiles(schemaPath, childPath, name, packageName string, embeds []string, outputDir string, useMessagePack bool) error {
+	outPath, err := buildStructFile(schemaPath, childPath, name, packageName, embeds, outputDir)
+	if err != nil {
+		return fmt.Errorf("failed to build struct file for %q: %v", name, err)
+	}
+
+	if useMessagePack {
+		msgp := "msgp"
+
+		if gopath := os.Getenv("GOPATH"); gopath != "" {
+			msgp = filepath.Join(gopath, "bin", msgp)
+		}
+
+		if err = exec.Command(msgp, "-file", outPath).Run(); err != nil {
```

Comment:
Marcelo discovered today that running msgp on a single file does not work if there are structs spread across multiple files. That means you will have to run it on the directory once after all the files are generated.

Reply:
Yes, Marcelo and I encountered this problem on Friday, but we didn't know how to resolve it, so I suggested he ask for your input.  It didn't occur to us to try using a directory as the `-file` argument, since the name of the flag implied that it would only work on a single file.  Running msgp on the entire directory at once fixes the problem with assets and draft assets, and that's what I've done with the rewrite.